File: javascript/app/apis/MembersApi.js
var swagger = require("swagger-node-express");
var url = require("url");
var errors = swagger.errors;
var params = swagger.params;

/* add model includes */

function writeResponse (response, data) {
  response.header('Access-Control-Allow-Origin', "*");
  response.header("Access-Control-Allow-Methods", "GET, POST, DELETE, PUT");
  response.header("Access-Control-Allow-Headers", "Content-Type");
  response.header("Content-Type", "application/json; charset=utf-8");
  response.send(JSON.stringify(data));
}

exports.models = models = require("../models.js");

exports.index = {
  'spec': {
    "description" : "Operations about pets",
    "path" : "api/advisors/{advisor_id}/members.json",
    "notes" : "This lists all the active members for a given advisor",
    "summary" : "Read member lead data",
    "method": "GET",
    "params" : [
      params.query("vestorly-auth", "Authentication token", "", true, false, ""),
    
      params.query("search", "Fuzzy search by name", "", false, false, ""),
    
      params.query("sorted_by", "Sort by: first_name, last_name, content_views, shares, posts_read, signed_up, last_active", "", false, false, ""),
    
      params.query("sorted_order", "Sort order: asc, desc", "", false, false, "")
    ].concat([
      params.path("advisor_id", "Id of Advisor"),
    ]).concat([]).concat([]),
    
    
    "type" : "Void",
    
    "responseMessages" : [errors.invalid('id'), errors.notFound('Void')],
    "nickname" : "index"
  },
  'action': function (req,res) {
    
    writeResponse(res, {message: "how about implementing index as a GET method?"});    
  }
};
exports.create = {
  'spec': {
    "description" : "Operations about pets",
    "path" : "api/advisors/{advisor_id}/members.json",
    "notes" : "This creates a member in the database",
    "summary" : "Create a member",
    "method": "POST",
    "params" : [
      params.query("vestorly-auth", "Authentication token", "", true, false, ""),
    ].concat([
      params.path("advisor_id", "Id of advisor"),
    ]).concat([]).concat([]),
    
    
    "type" : "Void",
    
    "responseMessages" : [errors.invalid('id'), errors.notFound('Void')],
    "nickname" : "create"
  },
  'action': function (req,res) {
    
    writeResponse(res, {message: "how about implementing create as a POST method?"});    
  }
};
exports.show = {
  'spec': {
    "description" : "Operations about pets",
    "path" : "api/advisors/{advisor_id}/members/{id}.json",
    "notes" : "This retrieves a specifp ic member object",
    "summary" : "Read a specific member",
    "method": "GET",
    "params" : [
      params.query("vestorly-auth", "Authentication token", "", true, false, ""),
    ].concat([
      params.path("advisor_id", "Id of advisor"),
    
      params.path("id", "Id of member")
    ]).concat([]).concat([]),
    
    
    "type" : "Void",
    
    "responseMessages" : [errors.invalid('id'), errors.notFound('Void')],
    "nickname" : "show"
  },
  'action': function (req,res) {
    
    writeResponse(res, {message: "how about implementing show as a GET method?"});    
  }
};
exports.update = {
  'spec': {
    "description" : "Operations about pets",
    "path" : "api/advisors/{advisor_id}/members/{id}.json",
    "notes" : "This updates a member in the database",
    "summary" : "Update a member",
    "method": "PUT",
    "params" : [
      params.query("vestorly-auth", "Authentication token", "", true, false, ""),
    ].concat([
      params.path("advisor_id", "Id of advisor"),
    
      params.path("id", "Id of member"),
    ]).concat([]).concat([]),
    
    
    "type" : "Void",
    
    "responseMessages" : [errors.invalid('id'), errors.notFound('Void')],
    "nickname" : "update"
  },
  'action': function (req,res) {
    
    writeResponse(res, {message: "how about implementing update as a PUT method?"});    
  }
};
exports.destroy = {
  'spec': {
    "description" : "Operations about pets",
    "path" : "api/advisors/{advisor_id}/members/{id}.json",
    "notes" : "This deletes a member in the database",
    "summary" : "Deletes a member",
    "method": "DELETE",
    "params" : [
      params.query("vestorly-auth", "Authentication token", "", true, false, ""),
    ].concat([
      params.path("advisor_id", "Id of advisor"),
    
      params.path("id", "Id of member")
    ]).concat([]).concat([]),
    
    
    "type" : "Void",
    
    "responseMessages" : [errors.invalid('id'), errors.notFound('Void')],
    "nickname" : "destroy"
  },
  'action': function (req,res) {
    
    writeResponse(res, {message: "how about implementing destroy as a DELETE method?"});    
  }
};

File: v1/scalatra/src/main/scala/com/wordnik/client/api/MembersApi.scala
package com.wordnik.client.api

import com.wordnik.client.model.Void

import java.io.File

import org.scalatra.{ TypedParamSupport, ScalatraServlet }
import org.scalatra.swagger._
import org.json4s._
import org.json4s.JsonDSL._
import org.scalatra.json.{ JValueResult, JacksonJsonSupport }
import org.scalatra.servlet.{FileUploadSupport, MultipartConfig, SizeConstraintExceededException}

import scala.collection.JavaConverters._

class MembersApi (implicit val swagger: Swagger) extends ScalatraServlet 
    with FileUploadSupport
    with JacksonJsonSupport
    with SwaggerSupport {
  protected implicit val jsonFormats: Formats = DefaultFormats

  protected val applicationDescription: String = "MembersApi"
  override protected val applicationName: Option[String] = Some("Members")

  before() {
    contentType = formats("json")
    response.headers += ("Access-Control-Allow-Origin" -> "*")
  }
  

  val indexOperation = (apiOperation[Void]("index")
      summary "Read member lead data"
      parameters(
        queryParam[String]("vestorly-auth").description("")
        
        
        
        
        
        ,
        
        pathParam[String]("advisor_id").description("")
        
        
        
        
        ,
        queryParam[String]("search").description("").optional
        
        
        
        
        
        ,
        queryParam[String]("sorted_by").description("").optional
        
        
        
        
        
        ,
        queryParam[String]("sorted_order").description("").optional
        
        
        
        
        
        
        )
  )

  get("api/advisors/{advisor_id}/members.json",operation(indexOperation)) {
    
    
    
    

    
      
      val vestorly-auth = params.getAs[String]("vestorly-auth")
      
    

    

    

    
    
    println("vestorly-auth: " + vestorly-auth)
  
    
    
    
      val advisor_id = params.getOrElse("advisor_id", halt(400))
    

    

    

    

    
    
    println("advisor_id: " + advisor_id)
  
    
    
    

    
      
      val search = params.getAs[String]("search")
      
    

    

    

    
    
    println("search: " + search)
  
    
    
    

    
      
      val sorted_by = params.getAs[String]("sorted_by")
      
    

    

    

    
    
    println("sorted_by: " + sorted_by)
  
    
    
    

    
      
      val sorted_order = params.getAs[String]("sorted_order")
      
    

    

    

    
    
    println("sorted_order: " + sorted_order)
  
  }

  

  val createOperation = (apiOperation[Void]("create")
      summary "Create a member"
      parameters(
        queryParam[String]("vestorly-auth").description("")
        
        
        
        
        
        ,
        
        pathParam[String]("advisor_id").description("")
        
        
        
        
        ,
        
        
        
        
        formParam[String]("member").description("")
        
        
        )
  )

  post("api/advisors/{advisor_id}/members.json",operation(createOperation)) {
    
    
    
    

    
      
      val vestorly-auth = params.getAs[String]("vestorly-auth")
      
    

    

    

    
    
    println("vestorly-auth: " + vestorly-auth)
  
    
    
    
      val advisor_id = params.getOrElse("advisor_id", halt(400))
    

    

    

    

    
    
    println("advisor_id: " + advisor_id)
  
    
    
    

    

    

    
      val member = params.getAs[String]("member")
    

    
    
    println("member: " + member)
  
  }

  

  val showOperation = (apiOperation[Void]("show")
      summary "Read a specific member"
      parameters(
        queryParam[String]("vestorly-auth").description("")
        
        
        
        
        
        ,
        
        pathParam[String]("advisor_id").description("")
        
        
        
        
        ,
        
        pathParam[String]("id").description("")
        
        
        
        
        
        )
  )

  get("api/advisors/{advisor_id}/members/{id}.json",operation(showOperation)) {
    
    
    
    

    
      
      val vestorly-auth = params.getAs[String]("vestorly-auth")
      
    

    

    

    
    
    println("vestorly-auth: " + vestorly-auth)
  
    
    
    
      val advisor_id = params.getOrElse("advisor_id", halt(400))
    

    

    

    

    
    
    println("advisor_id: " + advisor_id)
  
    
    
    
      val id = params.getOrElse("id", halt(400))
    

    

    

    

    
    
    println("id: " + id)
  
  }

  

  val updateOperation = (apiOperation[Void]("update")
      summary "Update a member"
      parameters(
        queryParam[String]("vestorly-auth").description("")
        
        
        
        
        
        ,
        
        pathParam[String]("advisor_id").description("")
        
        
        
        
        ,
        
        pathParam[String]("id").description("")
        
        
        
        
        ,
        
        
        
        
        formParam[String]("member").description("")
        
        
        )
  )

  put("api/advisors/{advisor_id}/members/{id}.json",operation(updateOperation)) {
    
    
    
    

    
      
      val vestorly-auth = params.getAs[String]("vestorly-auth")
      
    

    

    

    
    
    println("vestorly-auth: " + vestorly-auth)
  
    
    
    
      val advisor_id = params.getOrElse("advisor_id", halt(400))
    

    

    

    

    
    
    println("advisor_id: " + advisor_id)
  
    
    
    
      val id = params.getOrElse("id", halt(400))
    

    

    

    

    
    
    println("id: " + id)
  
    
    
    

    

    

    
      val member = params.getAs[String]("member")
    

    
    
    println("member: " + member)
  
  }

  

  val destroyOperation = (apiOperation[Void]("destroy")
      summary "Deletes a member"
      parameters(
        queryParam[String]("vestorly-auth").description("")
        
        
        
        
        
        ,
        
        pathParam[String]("advisor_id").description("")
        
        
        
        
        ,
        
        pathParam[String]("id").description("")
        
        
        
        
        
        )
  )

  delete("api/advisors/{advisor_id}/members/{id}.json",operation(destroyOperation)) {
    
    
    
    

    
      
      val vestorly-auth = params.getAs[String]("vestorly-auth")
      
    

    

    

    
    
    println("vestorly-auth: " + vestorly-auth)
  
    
    
    
      val advisor_id = params.getOrElse("advisor_id", halt(400))
    

    

    

    

    
    
    println("advisor_id: " + advisor_id)
  
    
    
    
      val id = params.getOrElse("id", halt(400))
    

    

    

    

    
    
    println("id: " + id)
  
  }

}